semana8/insta-lab-hooks/src/components/Post/Post.js

```diff
@@ -0,0 +1,116 @@
+import React, { useState } from 'react'
+import { PostContainer, PostHeader, UserPhoto, PostPhoto, PostFooter, CommentContainer } from './styles'
+
+import IconeComContador from '../IconeComContador/IconeComContador'
+import SecaoComentario from '../SecaoComentario/SecaoComentario'
+
+import iconeCoracaoBranco from '../../img/favorite-white.svg'
+import iconeCoracaoPreto from '../../img/favorite.svg'
+import iconeComentario from '../../img/comment_icon.svg'
+
+const Post = (props) => {
+
+  const [curtido, setCurtido] = useState(false);
+
+  const [numeroCurtidas, setnumeroCurtidas] = useState(1);
+
+  const [comentando, setComentando] = useState(false);
+
+  const [numeroComentarios, setNumeroComentarios] = useState(1);
+
+  const [comentario, setComentario] = useState([]);
+  
+
+  const onClickCurtida = () => {
+if (curtido) {
+  setCurtido (!curtido)
+  setnumeroCurtidas(numeroCurtidas - 1)
+  this.setState({
```

Comment:
No componente funcional não tem mais this.setState! Ali em cima você já setou o estado certinho com o setCurtido e o setNumeroCurtidas, só apagar das linhas 28 até a 30